cmd/operator-sdk/printdeps/cmd.go

```diff
@@ -51,39 +48,50 @@ func printDepsFunc(cmd *cobra.Command, args []string) error {
 	if len(args) != 0 {
 		return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
 	}
+	projutil.MustInProjectRoot()
 
-	if err := printDeps(asFile); err != nil {
-		return fmt.Errorf("print deps failed: (%v)", err)
+	if err := printDeps(depManager); err != nil {
+		return fmt.Errorf("print deps failed: %v", err)
 	}
 	return nil
 }
 
-func printDeps(asFile bool) error {
+func printDeps(depManager string) (err error) {
 	// Make sure the project has a dep manager file.
-	mt, err := projutil.GetDepManagerType()
-	if err != nil {
-		return err
+	mt := projutil.DepManagerType(depManager)
+	if mt == "" {
+		mt, err = projutil.GetDepManagerType()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Can you add a comment here that we're checking the depManager flag type first and then falling back to auto detecting the dep manager type?

At first glance I didn't understand why we're checking the same thing twice since `DepManagerType()` `GetDepManagerType()` seem pretty close.

Reply:
Agreed, I had the same thought on my initial read.